i18n-helpers/src/bin/mdbook-xgettext.rs

```diff
@@ -93,9 +94,15 @@ fn create_catalog(ctx: &RenderContext) -> anyhow::Result<Catalog> {
                 Some(path) => ctx.config.book.src.join(path),
                 None => continue,
             };
-            for (lineno, msgid) in extract_messages(&chapter.content) {
+            for (lineno, extracted) in extract_messages(&chapter.content) {
+                let msgid = extracted.message;
                 let source = format!("{}:{}", path.display(), lineno);
-                add_message(&mut catalog, &msgid, &source);
+                add_message(
+                    &mut catalog,
+                    &msgid,
+                    &source,
+                    &extracted.translator_comment.unwrap_or(String::from("")),
```

Comment:
Try using

```rust
extracted.translator_comment.as_deref().unwrap_or_default()
```

instead. [Option::as_deref](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.as_deref) is one of those weird methods which you only see in Rust, it turns `Option<String>` into `Option<&str>`, after which `unwrap_or_default()` can give you the string or `""`.

In this case it'll probably generate the same code: `String::from("")` is the same as `String::new()` (I'm not good at reading assembly, but a quick check with [Godbolt](https://rust.godbolt.org/z/db5o8MasG) shows this) and `String::new()` does not allocate.

Reply:
Put differently, `&... String::from("...")` looks strange: you're constructing a `String` onto to turn it into a `&str` on the same line. That's what made me question this little line :slightly_smiling_face:

Reply:
Sounds good; switched to get_deref().